pytest_fixtures/component/discovery.py

```diff
@@ -0,0 +1,47 @@
+from copy import copy
+
+import pytest
+
+
+@pytest.fixture(scope='module')
+def module_discovery_sat(
+    module_provisioning_rhel_content, module_target_sat, module_sca_manifest_org, module_location
+):
+    """Creates a Satellite with discovery installed and configured"""
+    sat = module_target_sat
+    # Register to CDN and install discovery image
+    sat.register_to_cdn()
+    sat.execute('yum -y --disableplugin=foreman-protector install foreman-discovery-image')
+    sat.unregister()
+    # Symlink image so it can be uploaded for KEXEC
+    disc_img_path = sat.execute(
+        'find /usr/share/foreman-discovery-image -name "foreman-discovery-image-*.iso"'
+    )
+    disc_img_name = disc_img_path.stdout.split("/")[-1]
+    sat.execute(f'ln -s {disc_img_path} /var/www/html/pub/{disc_img_name}')
+    # Change 'Default PXE global template entry'
+    pxe_entry = sat.api.Setting().search(query={'search': 'Default PXE global template entry'})[0]
+    if pxe_entry.value != "discovery":
+        pxe_entry.value = "discovery"
+        pxe_entry.update(['value'])
+    # Build PXE default template to get default PXE file
+    sat.api.ProvisioningTemplate().build_pxe_default()
+    # Get default settings values
+    default_disco_settings = {
+        i.name: i for i in sat.api.Setting().search(query={'search': 'name~discovery'})
+    }
+
+    # Update discovery taxonomies settings
+    discovery_loc = copy(default_disco_settings['discovery_location'])
+    discovery_loc.value = module_location.name
+    discovery_loc.update(['value'])
+    discovery_org = copy(default_disco_settings['discovery_organization'])
+    discovery_org.value = module_sca_manifest_org.name
+    discovery_org.update(['value'])
```

Comment:
Looping and then again lookup by their name doesnt sounds like a good strategy, you can directly lookup by name of setting.

Check you can also use settings fixture here for multiple settings changes or we can update that fixture to consume multiple settings update.